### PR TITLE
fix: [PIE-6419]: fixed scope change so that repo resets

### PR DESCRIPTION
**Pull request by sumanmabd**
### Summary

Repo selected in the filter should be resetted if scope of the template is changed in selection page as shwon below.
#### Screenshots
image

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by sumanmabd** [comment]
retrigger cypress-cd

**Comment by sumanmabd** [comment]
retrigger cypress-cd

File: src/modules/72-templates-library/components/TemplateSelector/TemplateSelectorLeftView/TemplateSelectorLeftView.tsx [hunk 1]
```diff
@@ -203,6 +203,10 @@ export const TemplateSelectorLeftView: React.FC<TemplateSelectorLeftViewProps> =
     }
   }, [loading])
 
+  useEffect(() => {
```
Comment (chetanvekariya-harness) [inline src/modules/72-templates-library/components/TemplateSelector/TemplateSelectorLeftView/TemplateSelectorLeftView.tsx:206]: Can we do this in the onChange handle of scope ?

Comment (sumanmabd) [reply src/modules/72-templates-library/components/TemplateSelector/TemplateSelectorLeftView/TemplateSelectorLeftView.tsx]: Sure

**Review by samarth-sharma31 — APPROVED** [review]

**Comment by sumanmabd** [comment]
retrigger cypress-cv

**Review by udayvunnam — APPROVED** [review]

**Pull request merged**